include/common/core/freertos_synchronization.hpp

```diff
@@ -61,4 +61,19 @@ class FreeRTOSMutexFromISR {
     StaticSemaphore_t static_data{};
 };
 
+class FreeRTOSCriticalSection {
+  public:
+    FreeRTOSCriticalSection() {}
+    FreeRTOSCriticalSection(const FreeRTOSCriticalSection &) = delete;
+    FreeRTOSCriticalSection(const FreeRTOSCriticalSection &&) = delete;
+    FreeRTOSCriticalSection &operator=(const FreeRTOSCriticalSection &) =
+        delete;
+    FreeRTOSCriticalSection &&operator=(const FreeRTOSCriticalSection &&) =
+        delete;
+
+    void acquire() { taskENTER_CRITICAL(); }
```

Comment:
Are we purposefully trying to disable interrupts as well here?

Reply:
In this case, yes. The socketcan transport classes uses a blocking read and needs to not have it be interrupted by FreeRTOS.

I don't know of a use case for this class on the firmware.